### PR TITLE
feat(server): support changing MaxGRPCMessageSize using env variable

### PR DESCRIPTION
**Pull request by dpeer6**
…AGE_SIZE to argo-server

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

Tips:

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it does not need to pass
* Sign-off your commits to pass the DCO check: `git commit --signoff`.
* Run `make pre-commit -B` to fix codegen or lint problems. 
* Say how how you tested your changes. If you changed the UI, attach screenshots.

### CODE REVIEW

**Comment by dpeer6** [comment]
@alexec
This is the updated PR for the MaxGRPCMessageSize override using env varibale to argo-server deployment

**Review by alexec — APPROVED** [review]

**Review by alexec — CHANGES_REQUESTED** [review]
```
Documented variable GRPC_MESSAGE_SIZE in docs/environment-variables.md is not used anywhere
```

This is a bug in check-env-doc.sh, it does not include `./server`. Should be easy to fix and run.

**Review by alexec — APPROVED** [review]

**Comment by dpeer6** [comment]
I updated hack/check-env-doc.sh and now the test passes.. for me at least.

**Comment by dpeer6** [comment]
Hi @alexec 
Is there something else I can/should do to get this PR merged?
I see some tests FAILED but not sure it is related to that change

**Comment by alexec** [comment]
* `E2E Tests (test-cli, emissary, mysql)` this is not failing on `master` so could be a new bug (not sure how mind you!). 
* `Lint` this is failing because ` Error: File is not `goimports`-ed with -local github.com/argoproj/argo-workflows/ (goimports)`. Run `make lint`. Sometimes it is helps to delete any blank lines in your imports and this forces it to sort them.

**Comment by alexec** [comment]
So the build is failing for a good reason, because the CLI is now printing:

`time="2021-07-27T17:05:31Z" level=info msg="GRPC Server Max Message Size, MaxGRPCMessageSize, is set" `

So I was wrong. Don't print this is `init()`, I think you'll need to revert to your previous solution.

**Comment by dpeer6** [comment]
@alexec 
I moved the info logging message to the func that actually starts argo server.
hope that now it pass.
My local tests pass.

**Comment by alexec** [comment]
Merged at last!

File: docs/environment-variables.md [hunk 1]
```diff
@@ -35,6 +35,37 @@ Note that these environment variables may be removed at any time.
 | `WORKFLOW_GC_PERIOD` | `time.Duration` | `5m` | The periodicity for GC of workflows. |
 | `BUBBLE_ENTRY_TEMPLATE_ERR` | `bool` | `true` | Whether to bubble up template errors to workflow. |
 | `INFORMER_WRITE_BACK` | `bool` | `true` | Whether to write back to informer instead of catching up. |
+| `GRPC_MESSAGE_SIZE` | `string` | Use different GRPC Max message size for Argo server deployment (supporting huge workflows) |
+
+You can set environment variable for the argo-server deployment, for example:
```
Comment (agilgur5) [inline docs/environment-variables.md:40]: This is the section for the Controller, not the Server. It seems like a section for the Server did not exist at the time of this PR?

I moved it in #12148

**Pull request merged**